File: src/main/java/com/xeoneux/caravansary/entity/ReservationEntity.java
package com.xeoneux.caravansary.entity;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Entity
@Table(name = "Reservation")
public class ReservationEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotNull private LocalDate checkin;

    @NotNull private LocalDate checkout;

    @ManyToOne private RoomEntity roomEntity;

    public ReservationEntity() {
        super();
    }

    public ReservationEntity(LocalDate checkin, LocalDate checkout) {
        super();
        this.checkin = checkin;
        this.checkout = checkout;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDate getCheckin() {
        return checkin;
    }

    public void setCheckin(LocalDate checkin) {
        this.checkin = checkin;
    }

    public LocalDate getCheckout() {
        return checkout;
    }

    public void setCheckout(LocalDate checkout) {
        this.checkout = checkout;
    }

    public RoomEntity getRoomEntity() {
        return roomEntity;
    }

    public void setRoomEntity(RoomEntity roomEntity) {
        this.roomEntity = roomEntity;
    }
}

File: src/main/kotlin/com/xeoneux/caravansary/entity/ReservationEntity.kt
package com.xeoneux.caravansary.entity

import javax.persistence.*
import javax.validation.constraints.NotNull
import java.time.LocalDate

@Entity
@Table(name = "Reservation")
data class ReservationEntity(@Id
                             @GeneratedValue(strategy = GenerationType.AUTO)
                             var id: Long? = null,

                             @NotNull
                             var checkin: LocalDate? = null,

                             @NotNull
                             var checkout: LocalDate? = null,

                             @ManyToOne
                             var roomEntity: RoomEntity? = null) {

    constructor(checkin: LocalDate, checkout: LocalDate) : this(null, checkin, checkout)
}
